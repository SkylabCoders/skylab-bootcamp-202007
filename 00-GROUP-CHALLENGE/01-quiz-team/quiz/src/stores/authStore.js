import { EventEmitter } from 'events';
import dispatcher from '../dispatcher';
import actionTypes from './../actions/actionTypes';

const CHANGE_EVENT = 'change';
const _SESSION_DATA = {
    isLogged: false,
    userProfile: 'me@test.mail'
}

class AuthStore extends EventEmitter{

    addChangeListener(callback){
        this.on(CHANGE_EVENT, callback);
    }

    removeChangeListener(callback){
        this.removeListener(CHANGE_EVENT, callback);
    }

    emitChange(){
        this.emit(CHANGE_EVENT);
    }
    
    isLogged(){
        //console.log('STORE isLogged called from authStore with:', _SESSION_DATA.isLogged);
        return _SESSION_DATA.isLogged;
    }

    getUserProfile(){
        //console.log('STORE getUserProfile called from authStore with:', _SESSION_DATA.userProfile);
        return _SESSION_DATA.userProfile;
    }

}

const authStore = new AuthStore();

export default authStore;

dispatcher.register((action) => {
    switch (action.type){
        case actionTypes.LOGIN:
            //console.log('STORE registering action from dispatcher: LOGIN with data:', action.data);
            _SESSION_DATA.userProfile = action.data;
            _SESSION_DATA.isLogged = !!action.data;
            authStore.emitChange();
            break;
        case actionTypes.LOGOUT:
            //console.log('STORE registering action from dispatcher: LOGOUT');
            _SESSION_DATA.isLogged = false;
            _SESSION_DATA.userProfile = '';
            authStore.emitChange();
            break;
        default:
            break;
    }    
})

