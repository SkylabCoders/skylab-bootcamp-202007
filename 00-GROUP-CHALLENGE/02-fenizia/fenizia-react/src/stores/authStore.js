import EventEmiter from "events";
import dispatcher from "../appDispatcher";
import actionTypes from "../actions/actionTypes";

const CHANGE_EVENT = "change";
let _isLogged = false;
let _userProfile = null;
let _errorData = null;

class AuthStore extends EventEmiter {
  addChangeListener(callback) {
    this.on(CHANGE_EVENT, callback);
  }

  removeChangeListener(callback) {
    this.removeListener(CHANGE_EVENT, callback);
  }

  emitChange() {
    this.emit(CHANGE_EVENT);
  }

  isLogged() {
    return _isLogged;
  }

  getUserProfile() {
    return _userProfile;
  }
  getSignError() {
    return _errorData;
  }
}

const authStore = new AuthStore();

dispatcher.register((action) => {
  switch (action.type) {
    case actionTypes.LOGIN:
      _userProfile = action.user;
      _isLogged = !!action.user;
      authStore.emitChange();
      break;
    case actionTypes.LOGOUT:
      _isLogged = false;
      _userProfile = [];
      authStore.emitChange();
      break;
    case actionTypes.SIGN_IN:
      _errorData = action.data;
    default:
      break;
  }
});

export default authStore;
