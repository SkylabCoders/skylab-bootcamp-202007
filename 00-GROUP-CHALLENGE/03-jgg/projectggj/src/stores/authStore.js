import EventEmitter from 'events';
import actionTypes from '../actions/actionTypes';
import dispatcher from '../appDispatcher';

const CHANGE_EVENT = 'change';
let _isLogged = false;
let _userProfile = null;
let _userName = false;
let _photoUser = null;

class AuthStore extends EventEmitter {
    addChangeListener(callback) {
        this.on(CHANGE_EVENT, callback);
    }

    removeChangeListener(callback) {
        this.removeListener(CHANGE_EVENT, callback);
    }

    emitChange() {
        this.emit(CHANGE_EVENT);
    }

    isLogged() {
        return _isLogged;
    }
    getUserProfile() {
        return _userProfile;
    }
    getUserName() {
        return _userName;
    }
    getUserPhoto() {
        return _photoUser;
    }
}

const authStore = new AuthStore();

dispatcher.register((action) => {
    switch (action.type) {
        case actionTypes.LOGIN:
            _userProfile = action.data;
            _isLogged = !!action.data;
            _userName = action.data.user.displayName;
            _photoUser = action.data.user.photoURL;
            authStore.emitChange();
            break;
        case actionTypes.LOGOUT:
            _isLogged = false;
            _userProfile = null;
            authStore.emitChange();
            break;

        //CREATE.PROFILE doesn't need any data because is not sending anything, in fact you don't have to add any
        case actionTypes.CREATE_PROFILE:
            _userProfile = action.data;
            authStore.emitChange();
            break;
        default:
            return (`Action not recognized: ${action.type}`);
    }
});

export default authStore;
