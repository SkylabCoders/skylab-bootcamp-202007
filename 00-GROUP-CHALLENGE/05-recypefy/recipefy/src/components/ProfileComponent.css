.main__profile-component {
  display: none;
}

@media only screen and (min-width: 1024px) {
  .profile--preferences {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    -ms-flex-wrap: wrap;
        flex-wrap: wrap;
    -webkit-box-pack: space-evenly;
        -ms-flex-pack: space-evenly;
            justify-content: space-evenly;
    list-style: none;
  }
  .main__profile-component {
    background: #65dfc3;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    width: 20%;
    position: absolute;
    margin: 5% 2% 5% 2%;
    padding: 2%;
    border-radius: 25px;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    text-align-last: center;
  }
  .main__profile-component ul li {
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    padding: 5px;
  }
  .main__profile-component ul a {
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
  }
  .main__profile-component ul .main__profile-component--display-large {
    display: none;
  }
  .main__profile-component--title {
    margin: 20px 5px 5px 5px;
  }
  a ul {
    margin: auto;
  }
  .profile-img {
    border-radius: 100px;
    width: 150px;
    height: 150px;
    margin: 5%;
    border: 5px solid #0c0c0c;
  }
}
/*# sourceMappingURL=ProfileComponent.css.map */