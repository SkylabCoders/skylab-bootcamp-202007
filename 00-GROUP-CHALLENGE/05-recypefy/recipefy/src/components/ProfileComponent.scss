.main__profile-component {
	display: none;
}

@media only screen and (min-width: 1024px) {


	.profile--preferences {
		display: flex;
		flex-direction: column;
		flex-wrap: wrap;
		justify-content: space-evenly;
		list-style: none;

	}

	.main__profile-component {
		background: #65dfc3;
		display: flex;
		flex-direction: column;
		width: 20%;
		position: absolute;
		margin: 5% 2% 5% 2%;
		padding: 2%;
		border-radius: 25px;
		align-items: center;
		text-align-last: center;

		ul {
			li {
				align-items: center;
				padding: 5px;
			}

			a {
				align-items: center;
			}

			.main__profile-component--display-large {
				display: none;
			}
		}

	}

	.main__profile-component--title {
		margin: 20px 5px 5px 5px;
	}

	a {
		ul {
			margin: auto;
		}
	}

	.profile-img {
		border-radius: 100px;
		width: 150px;
		height: 150px;
		margin: 5%;
		border: 5px solid #0c0c0c;
	}
}