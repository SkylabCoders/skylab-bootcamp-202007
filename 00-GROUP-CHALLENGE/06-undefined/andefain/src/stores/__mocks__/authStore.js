const authStore = {
	logged: false,
	setLogged(val) {
		this.logged = val;
	},

	addChangeListener(callback) {},
	removeChangeListener(callback) {},
	isLogged() {
		return this.logged;
	}
};

export default authStore;
