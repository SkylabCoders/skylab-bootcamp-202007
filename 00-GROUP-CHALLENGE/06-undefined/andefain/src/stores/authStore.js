import EventEmitter from 'events';
import actionTypes from '../actions/actionTypes';
import dispatcher from '../dispatcher';

const CHANGE_EVENT = 'change';
let _isLogged = false;
let _userProfile = {
	displayName: 'John Doe',
	email: 'test@gmail.com',
	photoURL:
		'https://preview.keenthemes.com/metronic-v4/theme/assets/pages/media/profile/profile_user.jpg',
	movies: []
};
class AuthStore extends EventEmitter {
	addChangeListener(callback) {
		this.on(CHANGE_EVENT, callback);
	}

	removeChangeListener(callback) {
		this.removeListener(CHANGE_EVENT, callback);
	}

	emitChange() {
		this.emit(CHANGE_EVENT);
	}

	isLogged() {
		return _isLogged;
	}

	getUserProfile() {
		return _userProfile;
	}
}

const authStore = new AuthStore();
dispatcher.register((action) => {
	switch (action.type) {
		case actionTypes.LOGIN:
			_userProfile = action.data;
			_isLogged = !!action.data;
			authStore.emitChange();
			break;
		case actionTypes.LOGOUT:
			_isLogged = false;
			_userProfile = null;
			authStore.emitChange();
			break;
		default:
		//throw `The action type is unknown. action.type: ${action.type}`;
	}
});
export default authStore;
