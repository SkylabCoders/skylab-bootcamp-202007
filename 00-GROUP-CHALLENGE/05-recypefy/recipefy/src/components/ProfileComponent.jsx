import React, { useEffect, useState } from 'react';
import authStore from '../stores/AuthStore';
import './ProfileComponent.scss';

function ProfileComponent() {
	const [user, setUser] = useState(authStore.getUserProfile());
	const [userPhoto, setUserPhoto] = useState('');

	useEffect(() => {
		authStore.addChangeListener(onChange);
		if (!user) {
			setUser(authStore.getUserProfile());
		}
		if (user.user.photoURL) {
			setUserPhoto(user.user.photoURL);
		}
		return () => authStore.removeChangeListener;
	}, [user]);

	function onChange() {
		setUser(authStore.getUserProfile());
	}
	return (
		<aside className="main__profile-component">
			<h2 className="wrapper">Your information</h2>
			<div>
				{userPhoto && (
					<img alt="your avatar" src={userPhoto} className="profile-img"></img>
				)}
				{!userPhoto && (
					<img
						alt="your avatar"
						src="https://image.flaticon.com/icons/svg/843/843260.svg"
						className="profile-img"
					></img>
				)}
				<h3 className="main__profile-component--title">Preferences</h3>
				<ul className="profile--preferences">
					<li key="balanced">
						<p>Balanced</p>
						<img
							alt="icon preference"
							className="preferences__icon preference__balanced"
							src="https://image.flaticon.com/icons/png/512/30/30636.png"
						></img>
					</li>
					<li key="high-protein">
						<p>High Protein</p>
						<img
							alt="icon preference"
							className="preferences__icon"
							src="https://icon-library.com/images/protein-icon/protein-icon-0.jpg"
						></img>
					</li>
					<li key="low-fat">
						<p>Low Fat</p>
						<img
							alt="icon preference"
							className="preferences__icon"
							src="https://www.pinclipart.com/picdir/big/150-1505070_low-fat-or-low-carb-icon-clipart.png"
						></img>
					</li>
					<li key="low-carb">
						<p>Low Carb</p>
						<img
							alt="icon preference"
							className="preferences__icon"
							src="https://cdn2.iconfinder.com/data/icons/organic-food-1/24/Low_Carb-512.png"
						></img>
					</li>
					<li key="segan">
						<p>Vegan</p>
						<img
							alt="icon preference"
							className="preferences__icon"
							src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5b/Vegan_friendly_icon.svg/900px-Vegan_friendly_icon.svg.png"
						></img>
					</li>
					<li key="vegetarian">
						<p>Vegetarian</p>
						<img
							alt="icon preference"
							className="preferences__icon"
							src="https://img.icons8.com/plasticine/100/000000/vegetarian-food.png"
						></img>
					</li>
					<li key="sugar-concious">
						<p>Sugar Conscious</p>
						<img
							alt="icon preference"
							className="preferences__icon"
							src="https://www.footys.co.za/images/icon-sugar.png"
						></img>
					</li>
					<li key="peanut-free">
						<p>Peanut Free</p>
						<img
							alt="icon preference"
							className="preferences__icon preference__peanut-free"
							src="https://cdn0.iconfinder.com/data/icons/food-product-labels/128/peanut-free-512.png"
						></img>
					</li>
					<li key="tree-nut-freee">
						<p>Tree Nut Free</p>
						<img
							alt="icon preference"
							className="preferences__icon preference__tree-nut-free"
							src="https://cdn3.iconfinder.com/data/icons/food-allergens-3/77/allergens-tree-nut-free-512.png"
						></img>
					</li>
					<li key="alcohol-free">
						<p>Alcohol-Free</p>
						<img
							alt="icon preference"
							className="preferences__icon preference__alcohol-free"
							src="https://cdn3.iconfinder.com/data/icons/food-allergens-3/77/allergens-alcohol-free-512.png"
						></img>
					</li>
				</ul>
			</div>
		</aside>
	);
}

export default ProfileComponent;
