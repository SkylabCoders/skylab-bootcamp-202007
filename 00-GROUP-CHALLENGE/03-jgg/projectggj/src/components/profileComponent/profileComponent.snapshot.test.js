import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileComponent from './profileComponent';
import authStore from '../../stores/authStore';


describe('profileComponent', () => {
    const profileComponentTree = renderer.create(
        <Router>
            <ProfileComponent />
        </Router>
    );

    let instance;
    let component;
    let text;
    let myCallback;

    beforeEach(() => {
        myCallback = jest.fn();
        authStore.addChangeListener(myCallback);
    });
    afterEach(() => {
        authStore.removeChangeListener(myCallback);
    });

    it('should macth', () => {
        expect(profileComponentTree).toMatchSnapshot();
    });

    it('should display a paragraph', () => {
        instance = profileComponentTree.root;
        component = instance.findByType('p');
        text = component.children[0];

        expect(text).toEqual('Do you want to increase your victories?');
    });

    xit('should display a pic', () => {
        instance = profileComponentTree.root;
        component = instance.findByType('image');
        text = component.children[0];

        expect(text).toBeDefined();
    });


    it('should get name', () => {

        let user = false;
        function fn() {
            return authStore.getUserName();
        }

        expect(fn()).toEqual(user);
    });


});