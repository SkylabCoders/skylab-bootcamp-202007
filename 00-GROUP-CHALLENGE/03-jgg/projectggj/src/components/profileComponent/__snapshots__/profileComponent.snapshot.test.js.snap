// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profileComponent should macth 1`] = `
<div
  className="details-holder"
>
  <div
    className="flex-item"
  >
    <div
      className="row"
    >
      <div
        className="detail-container flex-item"
      >
        <div
          className="card hovercard"
        >
          <div
            className="photoProfile"
          />
          <div
            className="avatar"
          >
            <img
              alt="user profile"
              src="https://w7.pngwing.com/pngs/304/275/png-transparent-user-profile-computer-icons-profile-miscellaneous-logo-monochrome.png"
            />
          </div>
          <div
            className="info"
          >
            <div
              className="title"
            >
              <div>
                PROFILE
              </div>
            </div>
            <div
              className="profile-desc-content text-profile"
            >
              <div
                className="details-desc"
              >
                FULL NAME: 
                <span>
                  Undefined
                </span>
              </div>
              <div
                className="details-desc"
              >
                EMAIL: 
                <span>
                  Undefined
                </span>
              </div>
              <div
                className="details-desc"
              >
                RACE: 
                <span>
                  Human
                </span>
              </div>
              <div
                className="details-desc"
              >
                STATUS: 
                <span>
                  Alive
                </span>
              </div>
              <div
                className="details-desc"
              >
                PLANET: 
                <span>
                  Earth
                </span>
              </div>
              <div
                className="details-desc"
              >
                Nº OF VICTORIES: 
                <span>
                  0
                </span>
              </div>
              <div
                className="details-desc"
              />
              <div
                className="details-desc"
              >
                Nº OF DEFEATS: 
                <span>
                  0
                </span>
              </div>
              <p>
                Do you want to increase your victories?
              </p>
              <a
                href="/game/:enemy"
                onClick={[Function]}
              >
                Click here to fight!
              </a>
            </div>
            <div>
              <button
                className="'row figthlink btn'"
                onClick={[Function]}
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  Logout
                </a>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
